sysl2/sysl/examples/grammars/go/go.gen.g

```diff
@@ -5,7 +5,7 @@ ImportDecl: 'import' '(\n' ImportSpec* '\n)\n';
 ImportSpec: Import '\n';
 TopLevelDecl: Declaration | FunctionDecl | MethodDecl;
 Declaration: VarDecl | ConstDecl | StructType | InterfaceType ;
-StructType : 'type' StructName 'struct' '{\n' FieldDecl* '}\n';
+StructType : Comment? '\n' 'type' StructName 'struct' '{\n' FieldDecl* '}\n';
```

Comment:
Comments can probably go in many places -- it seems a bit strange to just have comment as an optional prefix of Struct in the grammar.


for example, it looks like Go's AST machinery has a map from AST nodes to comments:

https://golang.org/pkg/go/ast/#NewCommentMap

Reply:
This is used just to avoid inter warnings.

Reply:
Yes I agree. For the moment this is added to avoid linter complains.

Reply:
Comments are generally not part of the grammar. Have a look here: https://golang.org/ref/spec#Struct_types
We are doing the reverse of what generally grammars are used for ie. generating text instead of parsing them. So don't be surprised with these new constructs.

The point of this grammar is to assist you in producing the text in the structure that you want.

If you want to have multiple comments, for eg. struct comment, field comment, they will have their own place in the grammar.